src/unsafe-rust/unsafe-traits.md

```diff
@@ -8,36 +8,32 @@ Like with functions, you can mark a trait as `unsafe` if the implementation must
 guarantee particular conditions to avoid undefined behaviour.
 
 For example, the `zerocopy` crate has an unsafe trait that looks
-[something like this](https://docs.rs/zerocopy/latest/zerocopy/trait.AsBytes.html):
+[something like this](https://docs.rs/zerocopy/latest/zerocopy/trait.IntoBytes.html):
 
 ```rust,editable
-use std::mem::size_of_val;
-use std::slice;
+use std::{mem, slice};
 
 /// ...
 /// # Safety
 /// The type must have a defined representation and no padding.
-pub unsafe trait AsBytes {
+pub unsafe trait IntoBytes {
     fn as_bytes(&self) -> &[u8] {
-        unsafe {
-            slice::from_raw_parts(
-                self as *const Self as *const u8,
-                size_of_val(self),
-            )
-        }
+        let len = mem::size_of_val(self);
+        let slf: *const Self = self;
+        unsafe { slice::from_raw_parts(slf.cast::<u8>(), len) }
     }
 }
 
-// SAFETY: `u32` has a defined representation and no padding.
-unsafe impl AsBytes for u32 {}
+/// SAFETY: `u32` has no padding bytes.
```

Comment:
The defined representation is also important, and safety comments aren't usually doc comments.